diff --git a/src/assets_writer.cpp b/src/assets_writer.cpp
index 49ca8a9..38b01a9 100644
--- a/src/assets_writer.cpp
+++ b/src/assets_writer.cpp
@@ -45,7 +45,7 @@
 #include "tf2_eigen/tf2_eigen.hpp"
 #include "tf2_msgs/msg/tf_message.hpp"
 #include "tf2_ros/buffer.h"
-#include "urdf/model.h"
+#include "urdf/model.hpp"
 
 namespace cartographer_ros {
 namespace {
diff --git a/src/offline_node.cpp b/src/offline_node.cpp
index 94df3b0..1b67fd3 100644
--- a/src/offline_node.cpp
+++ b/src/offline_node.cpp
@@ -31,7 +31,7 @@
 #include "gflags/gflags.h"
 #include "rosgraph_msgs/msg/clock.hpp"
 #include "tf2_ros/static_transform_broadcaster.h"
-#include "urdf/model.h"
+#include "urdf/model.hpp"
 #include "rclcpp/exceptions.hpp"
 #include <regex>
 #include <string>
diff --git a/src/rosbag_validate_main.cpp b/src/rosbag_validate_main.cpp
index 1c7ba2a..0b55c6c 100644
--- a/src/rosbag_validate_main.cpp
+++ b/src/rosbag_validate_main.cpp
@@ -36,7 +36,7 @@
 #include "tf2_eigen/tf2_eigen.hpp"
 #include "tf2_msgs/msg/tf_message.hpp"
 #include "tf2_ros/buffer.h"
-#include "urdf/model.h"
+#include "urdf/model.hpp"
 #include <tf2/utils.h>
 
 DEFINE_string(bag_filename, "", "Bag to process.");
diff --git a/src/urdf_reader.cpp b/src/urdf_reader.cpp
index d2380cd..c092536 100644
--- a/src/urdf_reader.cpp
+++ b/src/urdf_reader.cpp
@@ -20,7 +20,7 @@
 #include <vector>
 
 #include "cartographer_ros/msg_conversion.h"
-#include "urdf/model.h"
+#include "urdf/model.hpp"
 
 namespace cartographer_ros {
 
