diff --git a/src/publisher.cpp b/src/publisher.cpp
index dd4f164..532cb7d 100644
--- a/src/publisher.cpp
+++ b/src/publisher.cpp
@@ -127,15 +127,15 @@ void Publisher::declareParameter(
     encoder_.setEncoder(v.get<std::string>());
     RCLCPP_INFO_STREAM(logger_, "using encoder: " << v.get<std::string>());
   } else if (n == "preset") {
-    encoder_.setPreset(v.get<std::string>());
+    encoder_.addAVOption(n, v.get<std::string>());
   } else if (n == "tune") {
-    encoder_.setTune(v.get<std::string>());
+    encoder_.addAVOption(n, v.get<std::string>());
   } else if (n == "delay") {
-    encoder_.setDelay(v.get<std::string>());
+    encoder_.addAVOption(n, v.get<std::string>());
   } else if (n == "crf") {
-    encoder_.setCRF(v.get<std::string>());
+    encoder_.addAVOption(n, v.get<std::string>());
   } else if (n == "pixel_format") {
-    encoder_.setPixelFormat(v.get<std::string>());
+    encoder_.setAVSourcePixelFormat(v.get<std::string>());
   } else if (n == "qmax") {
     encoder_.setQMax(v.get<int>());
   } else if (n == "bit_rate") {
