diff --git a/CMakeLists.txt b/CMakeLists.txt
index 521b07f..0624f3d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -20,6 +20,11 @@ rclcpp_components_register_node(${PROJECT_NAME}
   EXECUTOR MultiThreadedExecutor
 )
 
+target_compile_options(${PROJECT_NAME}_node
+  PRIVATE
+    -Wno-deprecated-declarations
+)
+
 if(BUILD_TESTING)
   function(add_testcase filepath)
     get_filename_component(filename ${filepath} NAME)
